Sources/DarkModeCore/DMTraitCollection.m

```diff
@@ -27,10 +27,16 @@ + (void)swizzleTraitCollectionDidChangeToDMTraitCollectionDidChangeWithBlock:(vo
     NSAssert(NO, @"Method not found for [%@ traitCollectionDidChange:]", NSStringFromClass(self));
 
   IMP imp = method_getImplementation(method);
-  class_replaceMethod(self, selector, imp_implementationWithBlock(^(NSObject *self, UITraitCollection *previousTraitCollection) {
+  class_replaceMethod(self, selector, imp_implementationWithBlock(^(id<UITraitEnvironment> self, UITraitCollection *previousTraitCollection) {
     // Call previous implementation
     ((void (*)(NSObject *, SEL, UITraitCollection *))imp)(self, selector, previousTraitCollection);
 
+    if (previousTraitCollection != nil && previousTraitCollection.userInterfaceStyle == self.traitCollection.userInterfaceStyle) {
+      // We only care about userInterfaceStyle change currently, so filter out
+      // calling without this specific change
+      return;
+    }
+
```

Comment:
Since the name of this method is `dmTraitCollectionDidChange`, I think we shouldn't filter anything here. It should be the responsibility of the developer who overrides this method to implement such logic.

Reply:
i would say since our API is called dmTraitCollectionDidChange, when other trait (that DMTraitCollection does not provide) changes, I don’t expect it to be called. Developers should still be handling the other changes in UIKit’s traitCollectionDidChange, but just need to remember call super.traitCollectionDidChange to ensure dmTraitCollectionDidChange is called. 

What do you think? @imWildCat @kylinroc

Reply:
I’m on @zhuorantan’s side, I’d prefer less code and logic since they are just maintenance debts.